File: parasmittal/DAY 12/Single Number II.cpp
class Solution {
public:
    int singleNumber(vector<int>& arr) {
       int result = 0; 
  int n=arr.size();
    int x, sum; 
    for (int i = 0; i < 32; i++) { 
        sum = 0; 
        x = (1 << i); 
        for (int j = 0; j < n; j++) { 
            if (arr[j] & x) 
                sum++; 
        } 
        if ((sum % 3) != 0) 
            result |= x; 
    } 
    return result; 
    }
};

File: sanket-dalvi/Day10/singleNumberII.py
from collections import Counter
from typing import List
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        if len(nums)==1: return nums[0]
        c = Counter(nums)
        for key in c:
            if c[key]==1:
                return key
        return -1


# Test
print(Solution().singleNumber([-2,-2,1,1,-3,1,-3,-3,-4,-2]))
# print(Solution().singleNumber([-2,1,1,-3,1,-3,-3,2,-2]))
print(Solution().singleNumber([0,1,0,1,0,1,99]))